readthedocs/search/views.py

```diff
@@ -86,21 +111,8 @@ def elastic_search(request, project_slug=None):
         project_obj = get_object_or_404(queryset, slug=project_slug)
         request_type = request.GET.get('type', 'file')
 
-    user_input = UserInput(
-        query=request.GET.get('q'),
-        type=request_type or request.GET.get('type', 'project'),
-        project=project_slug or request.GET.get('project'),
-        version=request.GET.get('version', LATEST),
-        taxonomy=request.GET.get('taxonomy'),
-        language=request.GET.get('language'),
-        role_name=request.GET.get('role_name'),
-        index=request.GET.get('index'),
-        publisher=request.GET.get('publisher'),
-        publisher_project=request.GET.getlist('publisher_project'),
-        sort=request.GET.get('sort'),
-        tags=request.GET.getlist('tags'),
-        page=request.GET.get('page'),
-    )
+    user_input = _get_user_input_namedtuple(request, request_type, project_slug)
```

Comment:
can we keep the current code structure and just add the "ifs"?

Reply:
It will be dirty like:
```
if request.GET.get('version'):
    UserInput = collections.namedtuple('UserInput', UserInput._fields + ('version',))
    user_input = UserInput(
        query=request.GET.get('q'),
        type=request_type or request.GET.get('type', 'project'),
        project=project_slug or request.GET.get('project'),
        version=request.GET.get('version'),
        taxonomy=request.GET.get('taxonomy'),
        language=request.GET.get('language'),
        role_name=request.GET.get('role_name'),
        index=request.GET.get('index'),
        publisher=request.GET.get('publisher'),
        publisher_project=request.GET.getlist('publisher_project'),
        sort=request.GET.get('sort'),
        tags=request.GET.getlist('tags'),
        page=request.GET.get('page'),
    )
else:
    UserInput = collections.namedtuple('UserInput', UserInput._fields + ('is_default',))
    user_input = UserInput(
        query=request.GET.get('q'),
        type=request_type or request.GET.get('type', 'project'),
        project=project_slug or request.GET.get('project'),
        is_default=True,
        taxonomy=request.GET.get('taxonomy'),
        language=request.GET.get('language'),
        role_name=request.GET.get('role_name'),
        index=request.GET.get('index'),
        publisher=request.GET.get('publisher'),
        publisher_project=request.GET.getlist('publisher_project'),
        sort=request.GET.get('sort'),
        tags=request.GET.getlist('tags'),
        page=request.GET.get('page'),
    )
```

Reply:
ok let me think